pkg/cache/memory/kvcache.go

```diff
@@ -2,22 +2,19 @@ package memory
 
 import (
 	"fmt"
-	"sync"
 	"time"
 
 	"github.com/pokt-network/poktroll/pkg/cache"
+	"github.com/puzpuzpuz/xsync/v4"
 )
 
 var _ cache.KeyValueCache[any] = (*keyValueCache[any])(nil)
 
 // keyValueCache provides a concurrency-safe in-memory key/value cache implementation.
 type keyValueCache[T any] struct {
 	config keyValueCacheConfig
-
-	// valuesMu is used to protect values AND valueHistories from concurrent access.
-	valuesMu sync.RWMutex
 	// values holds the cached values.
-	values map[string]cacheValue[T]
+	values *xsync.Map[string, cacheValue[T]]
```

Comment:
Could (should?) we create a TODO (or ticket) to move all map+mutex pair into xsync?

If so, can you create a ticket and leave a TODO somehwere in the code?